google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/CtxFunctions.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.conformance.retry;
+
+import static com.google.common.collect.Sets.newHashSet;
+
+import com.google.cloud.conformance.storage.v1.Resource;
+import com.google.cloud.storage.Acl;
+import com.google.cloud.storage.Acl.Role;
+import com.google.cloud.storage.Acl.User;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.conformance.retry.Functions.CtxFunction;
+import com.google.common.base.Joiner;
+import java.util.HashSet;
+
+final class CtxFunctions {
+
+  private static final class Util {
+    private static final CtxFunction blobIdAndBlobInfo =
+        (ctx, c) -> ctx.map(state -> state.with(BlobInfo.newBuilder(state.getBlobId()).build()));
+  }
+
+  static final class Local {
+    static final CtxFunction blobCopy =
+        (ctx, c) -> ctx.map(s -> s.withCopyDest(BlobId.of(c.getBucketName2(), c.getObjectName())));
+
+    static final CtxFunction bucketInfo =
+        (ctx, c) -> ctx.map(s -> s.with(BucketInfo.of(c.getBucketName())));
+    static final CtxFunction blobIdWithoutGeneration =
```

Comment:
Where are these actually used?

Reply:
These are used in the mappings and they function as a selective composable setup step since the API we have with JUnit is quite restrictive for this type of dynamic test generation we're doing.